### ISSUE TITLE
Add more JSON parseres

### ISSUE THREAD

**Issue by miloyip** [miloyip/nativejson-benchmark#9]
* [ ] https://github.com/nlohmann/json (gcc/clang only)
* [ ] https://github.com/pocoproject/poco/tree/develop/JSON/src

**Comment by miloyip** [comment]
sajson was added in sajson branch but it is unable to parse twitter.json. Reported an [issue](https://github.com/chadaustin/sajson/issues/7).

**Comment by miloyip** [comment]
JSON Voorhees fails to compile on 32-bit target. Reported as https://github.com/tgockel/json-voorhees/issues/53.

**Comment by miloyip** [comment]
ArduinoJson was added in https://github.com/miloyip/nativejson-benchmark/tree/ArduinoJson branch. But it cannot parse properly due to stack overflow. Reported as https://github.com/bblanchon/ArduinoJson/issues/65.

**Comment by miloyip** [comment]
nbsdx/SimpleJSON was added but there seems no way to iterate object and array. Reported as https://github.com/nbsdx/SimpleJSON/issues/1.

**Comment by miloyip** [comment]
jbson seems not working widely:

> Requires codecvt header (C++11, missing in gcc-4.8 & gcc-4.9 stdlib), various template aliases (C++14), probably more where gcc's stdlib is behind on the standard.
Known only to work with an up-to-date libc++ currently.

Thus not add it at this moment.

**Comment by miloyip** [comment]
ArduinoJson crashes when printing big number in conformance test. Reported as https://github.com/bblanchon/ArduinoJson/issues/67.

**Comment by xor-gate** [comment]
@miloyip as a notice the https://github.com/nlohmann/json has released v1.0.0, 16 days ago and would be worth to update.

**Comment by starwing** [comment]
@miloyip hello, I want to find a pure C, easy to use and intergrated json library, I found these:
  - https://github.com/sheredom/json.h
  - https://github.com/vurtun/mmx/blob/master/mm_json.h

maybe these can be added into benchmark, I found libraries [here](https://github.com/nothings/stb/blob/master/docs/other_libs.md), Thanks for these good results :-)

**Comment by denji** [comment]
Homepage: http://zserge.com/jsmn.html
Code: https://github.com/zserge/jsmn

minimalistic JSON parser in C, extremely small code footprint it’s just about 200 LOC.

![js_perf](https://cloud.githubusercontent.com/assets/3759759/12395657/af8830b4-be0b-11e5-8c40-55cd5b70f359.png)

**Comment by nlohmann** [comment]
@miloyip, thanks for updating nlohmann/json to 1.0.0. Is there a way to have a look at the new results?

**Comment by miloyip** [comment]
@nlohmann I will rerun the benchmarks after adding a few more libraries. You may also run it locally as well.

**Comment by xor-gate** [comment]
@miloyip is there no way to automate running the benchmarks with continues integration. E.g Travis CI ?

**Comment by miloyip** [comment]
@xor-gate Ideally of course that will be the best.
However, do you know any free service can provide stable performance environment?
Travis does not guarantee what build machine will be used, and how many concurrent builds (other users) will be running.

**Comment by xor-gate** [comment]
You are right about that, I have overseen this. But a colleague tried to run the tests but he could not get it to work. Maybe a alternative would be to have the test suite executable automaticly build under Docker. With a Dockerfile supplied, e.g debian jessie as base image then people can easy reproduce binaries and run on real metal.

**Comment by miloyip** [comment]
I understand there is difficult to build and run the tests. I faced these problems often as well.
The docker idea seems interesting. I have not tried docker before, not sure if there is potential problems.
Ideally the dockerfile is built in CI to make it always up-to-date. If you can help on trying this it will be great.

**Comment by xor-gate** [comment]
@miloyip I will open a separate issue for this so we when I need some help we can discus there. I will give it a shot this weekend.

**Comment by miloyip** [comment]
@starwing json.h done

**Comment by miloyip** [comment]
@starwing There are some issues when integrating MMX https://github.com/vurtun/mmx/issues/3

**Comment by miloyip** [comment]
@starwing Sigh... the author of mmx has removed the mm_json.h. I think I have not provoked him anyway https://github.com/vurtun/mmx/issues/3

**Comment by denji** [comment]
malloc() we can wrap [jemalloc](https://github.com/jemalloc/jemalloc), [scalloc](https://github.com/cksystemsgroup/scalloc), [hoard](https://github.com/emeryberger/Hoard) (`LD_PRELOAD`)
* https://github.com/cksystemsgroup/scalloc-artifact/blob/master/env.sh#L30
* https://github.com/cksystemsgroup/scalloc-artifact/blob/master/benchmarks/hoard/larson/larson.cpp#L47-L53

**Comment by nlohmann** [comment]
Hey @miloyip, today I released [version 1.1.0 of JSON for Modern C++](https://github.com/nlohmann/json/releases/tag/1.1.0). It would be great if you could update it in your benchmark.

This release also fixes some bugs that have been detected with your floating-point tests. :-)

**Comment by miloyip** [comment]
@nlohmann It's done. BTW, I just added a FAQ for updating libraries. Welcome for updating it via pull request. Good to know this test has helped improving your library :smile:

**Comment by starwing** [comment]
@miloyip so sad things :(

But still very thank you for the work!!

**Comment by giroudstephane** [comment]
@miloyip do you think [FlatBuffers](https://google.github.io/flatbuffers/flatbuffers_benchmarks.html) may have a place in your benchmark ?

**Comment by xor-gate** [comment]
In my opinion, this is a different format (just like MessagePack, Protocol Buffers, BSON) and should not be added to the nativejson-benchmark. As the name contains ... json.

**Comment by giroudstephane** [comment]
@xor-gate FlatBuffers can Parse JSON so maybe in some part of the benchmark this lib may be compared to the others

**Comment by xor-gate** [comment]
That make sense, I was not aware of that.

**Comment by user1095108** [comment]
Please add [js0n](https://github.com/quartzjer/js0n).

**Comment by mloskot** [comment]
Related requests #113 and  #118

**Comment by pps83** [comment]
Please add yyjson. This one seems to be the fastest json lib (faster than simdjson for large json files I process in my project)
